dist/iso_nso_calculator/index.js

```diff
@@ -0,0 +1,139 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.isoNsoCalculator = void 0;
+const read_ocf_package_1 = require("../read_ocf_package");
+const vesting_schedule_generator_1 = require("../vesting_schedule_generator");
+const addYearAndGrantId = (vestingSchedule, issuanceId) => {
+    return vestingSchedule.map((entry) => {
+        const date = new Date(entry.Date);
+        return Object.assign(Object.assign({}, entry), { Grant: issuanceId, Year: date.getFullYear() });
+    });
+};
+// Function to sum the amounts by year
+const sumByYear = (vestingSchedule) => {
+    // Create an empty object to store the sums by year
+    const result = {};
+    // Iterate through the array
+    vestingSchedule.forEach((entry) => {
+        if (entry["Event Type"] !== "Exercise") {
+            const year = entry.Year;
+            const amount = entry["Event Quantity"];
```

Comment:
what we're trying to capture here is actually the amount that became exercisable.

That will equal the `Event Quantity` in most cases, with the primary exception being early exercisable options.

(We've talked about supporting exercisability schedules separate from vesting schedules, but I'm not sure whether that's been deferred to future versions of the schema).

We could include logic in this module to determine the amount that first became exercisable, or we could add it as a field to the `vestingSchedule`.

Any thoughts?